src/detect-config.c

```diff
@@ -288,6 +288,12 @@ static int DetectConfigSetup (DetectEngineCtx *de_ctx, Signature *s, const char
         fd->scope = CONFIG_SCOPE_TX;
     }
 
+    if (fd->scope == CONFIG_SCOPE_TX) {
+        s->flags |= SIG_FLAG_APPLAYER;
+    } else if (fd->scope == CONFIG_SCOPE_FLOW) {
+        s->init_data->init_flags |= SIG_FLAG_INIT_FLOW;
```

Comment:
this will have side effects. The purpose of this flag is to track if the `flow:` keyword is in use in the sig. I think to trigger the `SignatureCreateMask` function itself currently.

Reply:
Ok cf https://github.com/OISF/suricata/pull/7080